tests/org/mozilla/gc/TestRuntime.java

```diff
@@ -0,0 +1,37 @@
+package org.mozilla.gc;
+
+import gnu.testlet.TestHarness;
+import gnu.testlet.Testlet;
+
+public class TestRuntime implements Testlet {
+    public int getExpectedPass() { return 5; }
+    public int getExpectedFail() { return 0; }
+    public int getExpectedKnownFail() { return 0; }
+
+    public void test(TestHarness th) {
+        long totalMemory = Runtime.getRuntime().totalMemory();
+        long freeMemory = Runtime.getRuntime().freeMemory();
+
+        System.out.println("freeMemory1: " + Runtime.getRuntime().freeMemory());
+
+        th.check(totalMemory > freeMemory, "Total memory is strictly greater than free memory");
+        th.check(freeMemory > 0, "Free memory is strictly greater than 0");
+
+        for (int i = 0; i < 1000; i++) {
+            long[] array = new long[100000];
+        }
+
+        System.out.println("freeMemory2: " + Runtime.getRuntime().freeMemory());
+
+        th.check(totalMemory, Runtime.getRuntime().totalMemory(), "Total memory doesn't change");
+        th.check(freeMemory > Runtime.getRuntime().freeMemory(), "Free memory decreases after allocating some objects");
```

Comment:
The problem with one small allocation is that you could end up freeing more memory than you've allocated (if a collection happens). A single big allocation should have the same effect of multiple small allocations, so I've done e976ccf.